website/docs/schema_evolution.md

```diff
@@ -22,21 +22,36 @@ the previous schema (e.g., renaming a column).
-| Add a custom nullable Hudi meta column, e.g. `_hoodie_meta_col`                  | Yes      | Yes     |                                                                                                                                                                                                                                                                                               |
-| Promote datatype from `int` to `long` for a field at root level                  | Yes      | Yes     | For other types, Hudi supports promotion as specified in [Avro schema resolution](http://avro.apache.org/docs/current/spec#Schema+Resolution).                                                                                                                                                |
-| Promote datatype from `int` to `long` for a nested field                         | Yes      | Yes     |
-| Promote datatype from `int` to `long` for a complex type (value of map or array) | Yes      | Yes     |                                                                                                                                                                                                                                                                                               |
-| Add a new non-nullable column at root level at the end                           | No       | No      | In case of MOR table with Spark data source, write succeeds but read fails. As a **workaround**, you can make the field nullable.                                                                                                                                                             |
-| Add a new non-nullable column to inner struct (at the end)                       | No       | No      |                                                                                                                                                                                                                                                                                               |
-| Change datatype from `long` to `int` for a nested field                          | No       | No      |                                                                                                                                                                                                                                                                                               |
-| Change datatype from `long` to `int` for a complex type (value of map or array)  | No       | No      |                                                                                                                                                                                                                                                                                               |
-
+The incoming schema will automatically have missing columns from the table schema added with null value
```

Comment:
I don't think we automatically add nulls right. user has to enable a config for that?

Reply:
how about the following

The incoming schema will automatically have missing columns added with null values from the table schema.
For this we need to enable the following config
`hoodie.write.handle.missing.cols.with.lossless.type.promotion`, otherwise the pipeline will fail. Note: This particular config will also do best effort to solve some of the backward incompatible
type promotions eg., 'long' to 'int'.